### PR TITLE
uninstaller: add ability to cleanup leftover resources with finalizers when longhorn uninstall stuck

### PR DESCRIPTION
**Pull request by shuo-wu**
https://github.com/rancher/longhorn/issues/462

### CODE REVIEW

File: deploy/uninstall/uninstall.yaml [hunk 1]
```diff
@@ -1,8 +1,58 @@
...
+metadata:
+  name: longhorn-uninstall-role
+rules:
+  - apiGroups:
+      - apiextensions.k8s.io
+    resources:
+      - customresourcedefinitions
+    verbs:
+      - "*"
+  - apiGroups: [""]
```
Comment (yasker) [inline deploy/uninstall/uninstall.yaml]: Looks like too many permissions? Is there a way to trim this down?

File: deploy/uninstall/uninstall.yaml [hunk 2]
```diff
@@ -20,8 +70,6 @@ spec:
         - --force
         env:
         - name: POD_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
+          value: longhorn-system
```
Comment (yasker) [inline deploy/uninstall/uninstall.yaml]: POD_NAMESPACE is normally pointed to the namespace pod is running. Change the name to e.g. `LONGHORN_NAMESPACE`.

Comment (shuo-wu) [reply deploy/uninstall/uninstall.yaml]: Modifying this leads to update manager image immediately. Is that ok?

Comment (yasker) [reply deploy/uninstall/uninstall.yaml]: Sure

**Comment by yasker** [comment]
LGTM

**Pull request merged**